### PR TITLE
Allow for UriInterface

### PR DESCRIPTION
**Pull request by Nyholm**
It is supported by `$this-> initialize()`

### CODE REVIEW

**Comment by Nyholm** [comment]
I see that there are some build failures but they are not related to this change.

File: src/Request.php [hunk 1]
```diff
@@ -24,7 +24,7 @@ class Request implements RequestInterface
     use MessageTrait, RequestTrait;
 
     /**
-     * @param null|string $uri URI for the request, if any.
+     * @param null|string|UriInterface $uri URI for the request, if any.
```
Comment (shadowhand) [inline src/Request.php]: As per https://github.com/Nyholm/zend-diactoros/blob/0eac0e66b09438b9a2a71d9071a466f0a55f3df2/src/RequestTrait.php#L55-L61

Comment (shadowhand) [inline src/Request.php]: Has been that way since [1.0.0](https://github.com/zendframework/zend-diactoros/blob/1.0.0/src/RequestTrait.php#L68-L80).

**Comment by weierophinney** [comment]
Thanks, @Nyholm

**Comment by Nyholm** [comment]
Thank you for merging.

**Pull request merged**